File: tests/fixtures/typescript/sample.ts
import { formatName as format } from "./shared";

export interface Greeter {
  greet(name: string): string;
}

export class UserService implements Greeter {
  public greet(name: string): string {
    return formatName(name);
  }
}

function formatName(name: string): string {
  return name.trim();
}

export const run = (name: string): string => {
  return format(name);
};

File: tests/fixtures/kotlin/sample.kt
package acme.sample

import kotlin.math.max
import acme.shared.Helper as SharedHelper
import acme.shared.*

interface Greeter {
    fun greet(name: String): String
}

open class BaseService

class UserService(private val helper: SharedHelper) : BaseService(), Greeter {
    val name: String = ""

    override fun greet(name: String): String {
        val normalized = format(name)
        println(max(normalized.length, 1))
        return helper.decorate(normalized)
    }

    private fun format(name: String): String = name.trim()
}

object ServiceFactory {
    fun create(): UserService {
        return UserService(SharedHelper())
    }
}
